src/query/src/sql/show_create_table.rs

```diff
@@ -46,13 +45,13 @@ fn create_sql_options(table_meta: &TableMeta, schema_options: Option<SchemaOptio
             write_buffer_size.to_string(),
         );
     }
-    if let Some(ttl) = table_opts.ttl {
-        options.insert(TTL_KEY.to_string(), format_duration(ttl).to_string());
-    } else if let Some(database_ttl) = schema_options.and_then(|o| o.ttl) {
-        options.insert(
-            TTL_KEY.to_string(),
-            format_duration(database_ttl).to_string(),
-        );
+    if let Some(ttl) = table_opts.ttl.and_then(|t| t.as_repr_opt()) {
+        options.insert(TTL_KEY.to_string(), ttl);
+    } else if let Some(database_ttl) = schema_options
+        .and_then(|o| o.ttl)
+        .and_then(|ttl| ttl.as_repr_opt())
```

Comment:
We can call `as_repr_opt` in the first `and_then`
```suggestion
        .and_then(|o| o.ttl.as_repr_opt())
```

Reply:
> We can call `as_repr_opt` in the first `and_then`

the first ttl is a Option so two and_then for flatten